### PR TITLE
STU3 Patient Export

### PR DESCRIPTION
**Pull request by kjmahalingam**
This PR adds FHIR STU3 export functionality from the Ruby version of Synthea into the Java version. The FHIR STU3 export now contains identifiers, race, ethnicity, and various other things that were previously not included.

These changes have not yet been validated by unit tests.

Items in other resources (encounter, procedure, medication, careplan) are yet to be implemented.

### CODE REVIEW

File: src/main/java/org/mitre/synthea/world/Demographics.java [hunk 1]
```diff
@@ -105,12 +105,79 @@ public String pickRace(Random random)
...
 		 */
-		
+
 		return raceDistribution.next(random);
 	}
-	
+
+	public String ethnicityFromRace(String race, Person person) {
+		// https://en.wikipedia.org/wiki/Demographics_of_Massachusetts#Race.2C_ethnicity.2C_and_ancestry
+		if (race == "white") {
```
Comment (dehall) [inline src/main/java/org/mitre/synthea/world/Demographics.java]: Compare strings using `str1.equals(str2)` not `str1 == str2` . Also this big if tower is the kind of thing that the RandomCollection is for. Ex:

to create it:
```
RandomCollection<String> whiteEthnicities = new RandomCollection<>();
whiteEthnicities.add(.228, "irish");
whiteEthnicities.add(.139, "italian");
[etc...]
```

Then to pick a result:
`whiteEthnicities.next(person.random);`

**Comment by jawalonoski** [comment]
Don't worry about code climate for now.

File: build.gradle [hunk 2]
```diff
@@ -18,7 +18,11 @@ apply plugin: 'eclipse'
 repositories {
     // Use jcenter for resolving your dependencies.
     // You can declare any Maven/Ivy/file repository here.
-    jcenter()
+    jcenter {
+
+        url "http://jcenter.bintray.com/"
```
Comment (jawalonoski) [inline build.gradle]: Why did you make this change?

Comment (kjmahalingam) [reply build.gradle]: I do not remember making this change. @dehall is this the change that we made as a temporary fix for getting this working on my local machine?

File: src/main/java/org/mitre/synthea/export/FhirStu3.java [hunk 3]
```diff
@@ -185,11 +367,9 @@ private static BundleEntryComponent basicInfo(Person person, Bundle bundle, long
...
 		// TODO: ruby version also has:
-		//  telecom, language, race, ethnicity, place of birth, mother's maiden name,
-		//  birth sex, prefix+suffix, marital status, multiple birth status,
-		//  identifiers (SSN, driver's lic, passport, fingerprint)
+		//  language, place of birth
 
-		String generatedBySynthea = 
+		String generatedBySynthea =
 				"Generated by <a href=\"https://github.com/synthetichealth/synthea\">Synthea</a>. "
 				+ "Version identifier: JAVA-0.0.0 . "
```
Comment (jawalonoski) [inline src/main/java/org/mitre/synthea/export/FhirStu3.java]: Don't hardcode the version info here.

File: src/main/java/org/mitre/synthea/export/FhirStu3.java [hunk 4]
```diff
@@ -204,18 +384,18 @@ private static BundleEntryComponent basicInfo(Person person, Bundle bundle, long
 		{
 			Extension dalyExtension = new Extension(SNOMED_URI + "/disability-adjusted-life-years");
```
Comment (jawalonoski) [inline src/main/java/org/mitre/synthea/export/FhirStu3.java]: QALY and DALY are not SNOMED codes.

File: src/main/java/org/mitre/synthea/export/FhirStu3.java (hunk 3 above)
Comment (kjmahalingam) [reply src/main/java/org/mitre/synthea/export/FhirStu3.java]: Okay. That change wasn't actually mine (the change listed was just a small styling change by my IDE). Where would you like me to move this?

File: build.gradle (hunk 2 above)
Comment (dehall) [reply build.gradle]: Yes this is a workaround to get past SSL inspection. There's a "better" way to do it that you have to re-do every time you update Java, so I prefer this. But it does require caution when committing code.

File: src/main/java/org/mitre/synthea/export/FhirStu3.java (hunk 3 above)
Comment (dehall) [reply src/main/java/org/mitre/synthea/export/FhirStu3.java]: In the ruby version the identifier is the the git commit hash, accessed by directly calling `git rev-parse HEAD` . I can take a look and see how to do it in Java

File: src/main/java/org/mitre/synthea/world/Demographics.java (hunk 1 above)
Comment (kjmahalingam) [reply src/main/java/org/mitre/synthea/world/Demographics.java]: Okay. I will implement these changes. You'll notice that I just implemented language, which is implemented very similarly to the if/else tower above. I'll make the changes you suggested here to that as well.

File: src/main/java/org/mitre/synthea/export/FhirStu3.java (hunk 4 above)
Comment (kjmahalingam) [reply src/main/java/org/mitre/synthea/export/FhirStu3.java]: It looks like these weren't implemented the same way in Ruby. I haven't been able to find any information about QALY and DALY extensions for FHIR online. Is there any way you want me to be implementing this?

Comment (jawalonoski) [reply src/main/java/org/mitre/synthea/export/FhirStu3.java]: Use a synthea extension URI.

File: src/main/resources/language_lookup.json [hunk 5]
```diff
@@ -0,0 +1,77 @@
+{
+  "english": {
+    "code": "en-US",
+    "system": "http://hl7.org/fhir/ValueSet/languages",
```
Comment (jawalonoski) [inline src/main/resources/language_lookup.json]: A `ValueSet` is not a system. If you do a Google search on that ValueSet, and look very carefully, you'll see that the `system` being used in that `ValueSet` is `"urn:ietf:bcp:47"`.

A `ValueSet` is a list of legal codes. Codes in that list can come from many `system` or terminologies. That language valueset consists of codes that come from that `"urn:ietf:bcp:47"` system. Make sense?

**Pull request merged**